### ISSUE TITLE
Progress does not decrease when bobber and fish are not overlapped

### ISSUE THREAD

**Issue by milchbrotchen** [cswendrowski/FoundryVTT-Obligatory-Fishing-Minigame#12]
![fish](https://user-images.githubusercontent.com/37191990/233756964-72b126a5-2bd2-4ad3-a338-925f5b8e7bf8.gif)

Tested with a new foundry world on D&D5e, no other mods.

**Comment by slave2midna** [comment]
Not sure if this project is abandoned - but I'm having this same issue under the same circumstances.

**Comment by milchbrotchen** [comment]
> Not sure if this project is abandoned - but I'm having this same issue under the same circumstances.

Yea I'm pretty sure it's abandoned, but I've edited fishingUi.mjs to fix the issue if you're still interested. It still works for Foundry v11 on DnD 3.1.2.

**Comment by slave2midna** [comment]
> > Not sure if this project is abandoned - but I'm having this same issue under the same circumstances.
> 
> Yea I'm pretty sure it's abandoned, but I've edited fishingUi.mjs to fix the issue if you're still interested. It still works for Foundry v11 on DnD 3.1.2.

Yes! I would LOVE that! If that's ok! That's the only issue I currently have with it. If I could fix that, it would be ok - I am also using v11.

**Comment by milchbrotchen** [comment]
[obligatory-fishing-minigame.zip](https://github.com/user-attachments/files/15889122/obligatory-fishing-minigame.zip)

The module, for simplicity. I also modified some variables to make the game play at a more Stardew valley pace. Unfortunately, I don't know how to fix the issue where you have to let the bobber fall all the way down if you want to lower it and any other bugs. If you're interested in editing these for yourself, see:
in fishingUi.mjs:
- Changed the progressPenalty from 0 to 1
- Movepersec from 1500 to 1000
- Added `this.rod.progressPenalty = Math.min(1, this.rod.progressPenalty - (shim.con * 0.1))` which basically means that the max penalty for not aligning with the fish is 1 progress per sec, which can be lowered by having better constitution. 
- Increased `this.fish.jumpRange = Math.min(70, this.fish.jumpRange - (shim.cha * 5));` for better fish jump range

And then in fishingSpotPageSheet.mjs, you can find the chances to roll fish, treasure, or trash in the second last block. I lowered the chance for treasure and increased junk, but maybe you might want different odds.

**Comment by slave2midna** [comment]
> [obligatory-fishing-minigame.zip](https://github.com/user-attachments/files/15889122/obligatory-fishing-minigame.zip)
> 
> The module, for simplicity. I also modified some variables to make the game play at a more Stardew valley pace. Unfortunately, I don't know how to fix the issue where you have to let the bobber fall all the way down if you want to lower it and any other bugs. If you're interested in editing these for yourself, see: in fishingUi.mjs:
> 
> * Changed the progressPenalty from 0 to 1
> * Movepersec from 1500 to 1000
> * Added `this.rod.progressPenalty = Math.min(1, this.rod.progressPenalty - (shim.con * 0.1))` which basically means that the max penalty for not aligning with the fish is 1 progress per sec, which can be lowered by having better constitution.
> * Increased `this.fish.jumpRange = Math.min(70, this.fish.jumpRange - (shim.cha * 5));` for better fish jump range
> 
> And then in fishingSpotPageSheet.mjs, you can find the chances to roll fish, treasure, or trash in the second last block. I lowered the chance for treasure and increased junk, but maybe you might want different odds.

This is wonderful! Amazing! -- It works much better than before in my game. Curious - if I wanted to increase he progress loss when not overlapping? What would I edit? We don't actually use a system that allows players to improve their performance - this is just a fun minigame for our fisherman. But as it stands, if he just leaves the green at the bottom - the progress will EVENTUALLY make it to the top because it doesn't decrease fast enough. 

https://github.com/cswendrowski/FoundryVTT-Obligatory-Fishing-Minigame/assets/69311684/b6e7afaf-5996-475d-b887-782114a0fb14

**Comment by slave2midna** [comment]
Nevermind - fixed it. Just increased progress to 2, and I removed the bonuses from character stats ( for our game it's not needed ) -- now it decreases fast enough. Thank you for the help!~
